### PR TITLE
Added `envelope*` methods to `Geometry`.

### PR DESCRIPTION
**Pull request by metasim**
- [X] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [X] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

### CODE REVIEW

File: gdal-sys/src/lib.rs [hunk 1]
```diff
@@ -6,3 +6,27 @@
...
+        Self {
+            MinX: 0.0,
+            MaxX: 0.0,
+            MinY: 0.0,
+            MaxY: 0.0,
+            MinZ: 0.0,
+            MaxZ: 0.0,
+        }
+    }
+}
```
Comment (metasim) [inline gdal-sys/src/lib.rs:32]: Is this controversial? Can we get bindgen to do it for us (on selected structs)?

File: src/vector/geometry.rs [hunk 2]
```diff
@@ -424,10 +425,30 @@ impl Geometry {
         Ok(unsafe { Geometry::with_c_geometry(new_c_geom, true) })
     }
 
+    /// Compute geometry area in square units of the spatial reference system in use.
+    ///
+    /// Supported for on`LinearRing`, `Polygon` or `MultiPolygon`.
```
Comment (lnicola) [inline src/vector/geometry.rs:430]:
```suggestion
    /// Supported for `LinearRing`, `Polygon` or `MultiPolygon`.
```

File: src/vector/geometry.rs [hunk 3]
```diff
@@ -424,10 +425,30 @@ impl Geometry {
...
+    ///
+    /// See: [`OGR_G_Area`](https://gdal.org/api/vector_c_api.html#_CPPv410OGR_G_Area12OGRGeometryH)
     pub fn area(&self) -> f64 {
         unsafe { gdal_sys::OGR_G_Area(self.c_geometry()) }
     }
 
+    /// Computes and returns the axis-aligned 2D bounding envelope for this geometry.
+    pub fn envelope(&self) -> Envelope {
+        let mut envelope = Envelope::default();
+        unsafe { gdal_sys::OGR_G_GetEnvelope(self.c_geometry(), &mut envelope) };
```
Comment (lnicola) [inline src/vector/geometry.rs:441]: Nit: do you want to give https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html a try here?

File: gdal-sys/src/lib.rs (hunk 1 above)
Comment (lnicola) [reply gdal-sys/src/lib.rs:32]: I don't think so, looks good to me.

File: src/vector/geometry.rs (hunk 3 above)
Comment (metasim) [reply src/vector/geometry.rs:443]: Do you mean using `MaybeUninit` instead of `Default`, and skip the initialization?

Comment (lnicola) [reply src/vector/geometry.rs:443]: Yes. It's not very important, just slightly nice to skip the initialization.

**Review by lnicola — APPROVED** [review]
The `MaybeUninit` uses look good, will merge in a day or so.

File: src/vector/geometry.rs [hunk 4]
```diff
@@ -424,10 +426,38 @@ impl Geometry {
...
+    /// See: [`OGR_G_GetEnvelope`](https://gdal.org/api/vector_c_api.html#_CPPv417OGR_G_GetEnvelope12OGRGeometryHP11OGREnvelope)
+    pub fn envelope(&self) -> Envelope {
+        let mut envelope = MaybeUninit::uninit();
+        unsafe {
+            gdal_sys::OGR_G_GetEnvelope(self.c_geometry(), envelope.as_mut_ptr());
+            envelope.assume_init()
+        }
+    }
+
+    /// Computes and returns the axis aligned 3D bounding envelope for this geometry.\
```
Comment (lnicola) [inline src/vector/geometry.rs:450]:
```suggestion
    /// Computes and returns the axis aligned 3D bounding envelope for this geometry.
```

File: src/vector/geometry.rs [hunk 5]
```diff
@@ -424,10 +426,38 @@ impl Geometry {
...
+    ///
+    /// See: [`OGR_G_GetEnvelope3D`](https://gdal.org/api/vector_c_api.html#_CPPv419OGR_G_GetEnvelope3D12OGRGeometryHP13OGREnvelope3D)
+    pub fn envelope3d(&self) -> Envelope3D {
```
Comment (lnicola) [inline src/vector/geometry.rs:453]: Nit:

```suggestion
    pub fn envelope_3d(&self) -> Envelope3D {
```

Not sure about this one, what do you think?

**Pull request merged**